freva-rest/src/freva_rest/tool_api/parse_config.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import tomllib
+from pathlib import Path
+from typing import Any, Union
+
+from pydantic import ValidationError
+
+from .tool_abstract import ParameterList
+
+
+def load_toml_to_dict(input_file:Union[str, Path]) -> dict:
+    """
+    Function to parse a given TOML file to a python dict object.
+
+    Args:
+        input_file (Union[str, Path]): Path to the TOML file to be parsed.
+
+    Returns:
+        dict: Dictionary containing keys and values of the TOML file.
+
+    Raises:
+        AssertionError: Raised when input_file does not exist or does not have a valid TOML file extension.
+        IOError: Raised when file cannot be opened in read-only mode. 
+        tomllib.TOMLDecodError: Raised when TOML file cannot be parsed by tomllib.
+    """
+    if isinstance(input_file, str): input_file = Path(input_file)
+    valid_toml_file_extensions = [".toml"]
+    assert input_file.exists(), f"Input file {input_file} cannot be found. Please make sure it exists and that you have read permissions for it." 
+    assert input_file.is_file() and input_file.suffix.lower() in valid_toml_file_extensions, f"Path {input_file} does not point to a recognised TOML file."

```

Comment:
ahhh, since these checks are part of the function's core validation logic, not debugging, i think you want to get them through this way:

```python
if not input_file.exists():
    raise FileNotFoundError(f"Input file {input_file} cannot be found. Please make sure it exists and that you have read permissions for it.")
    
if not input_file.is_file() or input_file.suffix.lower() not in valid_toml_file_extensions:
    raise ValueError(f"Path {input_file} does not point to a recognised TOML file.")
```

Reply:
Thanks, this file definitely needs more work. I will take your suggestions on board!